rotkehlchen/rotkehlchen.py

```diff
@@ -802,6 +802,10 @@ def query_balances(
             liabilities = {}
             log.error(f'Querying blockchain balances failed due to: {str(e)}')
 
+        # retrieve loopring balances if module is activated
+        if self.chain_manager.get_module('loopring'):
+            balances[str(Location.LOOPRING)] = self.chain_manager.get_loopring_balances()
```

Comment:
have you tested this? This will fail with:

```
2021-03-03T10:47:58.857Z: Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgreenlet.Greenlet.run
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 267, in _do_query_async
    result = getattr(self, command)(**kwargs)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 401, in _query_all_balances
    ignore_cache=ignore_cache,           
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/rotkehlchen.py", line 817, in query_balances
    assets_total_balance[asset] += balance
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/accounting/structures.py", line 65, in __add__
    other = _evaluate_balance_input(other, 'addition')
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/accounting/structures.py", line 210, in _evaluate_balance_input
    raise InputError(f'Found invalid dict object during Balance {operation}')
rotkehlchen.errors.InputError: Found invalid dict object during Balance addition
2021-03-03T10:22:45Z <Greenlet at 0x7efde2019ef0: <bound method RestAPI._do_query_async of <rotkehlchen.api.rest.RestAPI object at 0x7efde2faf110>>('_query_all_balances', 2, save_data=False, ignore_cache=False)> failed with InputError
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgreenlet.Greenlet.run
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 267, in _do_query_async
    result = getattr(self, command)(**kwargs)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 401, in _query_all_balances
    ignore_cache=ignore_cache,
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/rotkehlchen.py", line 817, in query_balances
    assets_total_balance[asset] += balance
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/accounting/structures.py", line 65, in __add__
    other = _evaluate_balance_input(other, 'addition')                                                                                                                                                                                                                           
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/accounting/structures.py", line 210, in _evaluate_balance_input
    raise InputError(f'Found invalid dict object during Balance {operation}')
rotkehlchen.errors.InputError: Found invalid dict object during Balance addition
2021-03-03T10:24:30Z <Greenlet at 0x7efde1e72290: <bound method RestAPI._do_query_async of <rotkehlchen.api.rest.RestAPI object at 0x7efde2faf110>>('_query_all_balances', 7, save_data=True, ignore_cache=True)> failed with InputError
```

The reason for this is that the function you made returns a mapping of `Dict[Address, Dict[Asset, Balance]]`.
And the `balances['loopring']` should contain a `Dict[Asset, Balance]`.

So in short you need to aggregate all loopring addresses from `get_loopring_balances()` function you introduced and return them in the format that would be easy to consume here during snapshotting.

Reply:
Yes, I just run the tests and didn't find anything. I was going to update tests to detect things like this. Will do this today

Reply:
after your recent changes it works.